src/program.cpp

```diff
@@ -86,8 +86,70 @@ static void print_program(const program& p, F print_func)
 program::program() : impl(std::make_unique<program_impl>()) {}
 
 program::program(program&&) noexcept = default;
-program& program::operator=(program&&) noexcept = default;
-program::~program() noexcept                    = default;
+program::~program() noexcept         = default;
+
+// copy constructor
+program::program(const program& p) { copy(p); }
+
+// copy assignment operator
+program& program::operator=(program p)
+{
+    std::swap(p.impl, this->impl);
+    return *this;
+}
+
+void program::copy(const program& p)
```

Comment:
This should be named `assign` not `copy`.

Reply:
fixed